wire.go

```diff
@@ -132,3 +133,22 @@ func Value(interface{}) ProvidedValue {
 func InterfaceValue(typ interface{}, x interface{}) ProvidedValue {
 	return ProvidedValue{}
 }
+
+// StructFields is a collection of the fields from a struct.
+type StructFields struct{}
+
+// FieldsOf extracts the structType into a list of types specified by fieldNames.
+// All types should be provided in the same provider set. The structType argument
```

Comment:
> All types should be provided in the same provider set.

Is this true or necessary? I think these can be split up. (I realize this is probably my fault for copy+pasting from the other issue.)

Reply:
Oops, this is not necessary.